browser/ipfs/ipfs_host_resolver.h

```diff
@@ -0,0 +1,67 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_BROWSER_IPFS_IPFS_HOST_RESOLVER_H_
+#define BRAVE_BROWSER_IPFS_IPFS_HOST_RESOLVER_H_
+
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "net/base/network_isolation_key.h"
+#include "services/network/public/cpp/resolve_host_client_base.h"
+#include "services/network/public/mojom/host_resolver.mojom.h"
+#include "services/network/public/mojom/network_context.mojom.h"
+
+namespace ipfs {
+
+// Resolves DNS TXT record for hosts. If prefix passed then
+// automatically adds it to the host.
+class IPFSHostResolver : public network::ResolveHostClientBase {
+ public:
+  explicit IPFSHostResolver(network::mojom::NetworkContext* network_context,
+                            const std::string& prefix = std::string());
+  ~IPFSHostResolver() override;
+
+  using HostTextResultsCallback =
+      base::OnceCallback<void(const std::string& host,
+                              const std::vector<std::string>& text_results)>;
+
+  void Resolve(const net::HostPortPair& host,
+               const net::NetworkIsolationKey& isolation_key,
+               net::DnsQueryType dns_query_type,
```

Comment:
looks like we're missing an include for `DnsQueryType` and `HostPortPair`

Reply:
a few others below too like base::Optional, etc... make sure to include what you use directly even if it's an override